Tensile/Source/lib/include/Tensile/PlaceholderLibrary.hpp

```diff
@@ -0,0 +1,142 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright 2022 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#pragma once
+
+#include <Tensile/MasterSolutionLibrary.hpp>
+#include <Tensile/SolutionLibrary.hpp>
+#include <Tensile/Tensile.hpp>
+
+#include <algorithm>
+
+namespace Tensile
+{
+
+    template <typename MyProblem, typename MySolution = typename MyProblem::Solution>
+    struct PlaceholderLibrary : public SolutionLibrary<MyProblem, MySolution>
+    {
+        mutable std::shared_ptr<SolutionLibrary<MyProblem, MySolution>> library;
+        mutable SolutionMap<MySolution>*                                solutions;
+        std::string                                                     filePrefix;
+        std::string                                                     libraryDirectory;
+
+        PlaceholderLibrary() = default;
+
+        bool loadPlaceholderLibrary() const
+        {
+#ifdef TENSILE_MSGPACK
+            std::string suffix = ".dat";
+#else
+            std::string suffix = ".yaml";
+#endif
```

Comment:
This has been causing lots of headaches in my local testing, and I don't like locking in the Tensile library to one format or the other at build time. The placeholder library already knows what file it's looking for, is there a good reason why you can't store the extension at the library level as well? (I sketched it out [here](https://github.com/benjaminulmer/Tensile/commit/d2c777b1ed8e0bd679836eff186619305929d75f) and it seems to work)

Reply:
Ah, yes I was testing with all the code objects preloaded, so that makes sense that I didn't encounter that.
If we want to avoid ugly extension stripping we could make the "value" have two (or more if needed) fields: filePrefix and then the extension for the library file.